src/overlays/actors/ovl_Bg_Hidan_Fwbig/z_bg_hidan_fwbig.c

```diff
@@ -21,29 +39,224 @@ const ActorInit Bg_Hidan_Fwbig_InitVars = {
     (ActorFunc)BgHidanFwbig_Update,
     (ActorFunc)BgHidanFwbig_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Fwbig/BgHidanFwbig_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Fwbig/BgHidanFwbig_Destroy.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x11, 0x00, 0x09, 0x20, COLSHAPE_CYLINDER },
+    { 0x00, { 0x20000000, 0x01, 0x04 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x19, 0x00, 0x01 },
+    { 30, 130, 0, { 0, 0, 0 } },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneScale, 1000, ICHAIN_STOP),
+};
+
+void BgHidanFwbig_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    BgHidanFwbig* this = THIS;
+    Player* player = PLAYER;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = 0xFF;
+    this->direction = (u16)(thisx->params >> 8);
+    thisx->params &= 0xFF;
+    if (this->direction != 0) {
+        this->actor.initPosRot.pos.x = 1560.0f;
+        this->actor.initPosRot.pos.z = 0.0f;
+        if (player->actor.posRot.pos.z > 300.0f) {
+            this->direction = -1;
+            this->actor.initPosRot.rot.y = this->actor.shape.rot.y = -0x4E38;
+        } else if (player->actor.posRot.pos.z < -300.0f) {
+            this->direction = 1;
+            this->actor.initPosRot.rot.y = this->actor.shape.rot.y = -0x31C8;
+        } else {
+            Actor_Kill(&this->actor);
+            return;
+        }
+        BgHidanFwbig_UpdatePosition(this);
+        Actor_SetScale(&this->actor, 0.15f);
+        this->collider.dim.height = 230;
+        this->actor.flags |= 0x10;
+        this->moveState = FWBIG_MOVE;
+        this->actionFunc = BgHidanFwbig_WaitForPlayer;
+        this->actor.posRot.pos.y = this->actor.initPosRot.pos.y - (2400.0f * this->actor.scale.y);
+    } else {
+        Actor_SetScale(&this->actor, 0.1f);
+        this->actionFunc = BgHidanFwbig_WaitForSwitch;
+    }
+}
+
+void BgHidanFwbig_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgHidanFwbig* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void BgHidanFwbig_UpdatePosition(BgHidanFwbig* this) {
+    s16 startAngle;
+
+    startAngle = this->actor.shape.rot.y + this->direction * -0x4000;
+
+    this->actor.posRot.pos.x = (Math_Sins(startAngle) * 885.4f) + this->actor.initPosRot.pos.x;
```

Comment:
personally id just put this assignment with the declaration and remove the extra line break